command/agent: LoadConfigDir behavior does not match docs.

```go
// LoadConfigDir loads all the configurations in the given directory
// in alphabetical order.
func LoadConfigDir(dir string) (*Config, error)
```

This function does not actually sort the files. It uses [Readdir](https://golang.org/pkg/os/#File.Readdir) to retrieve the list of files.

>Readdir reads the contents of the directory associated with file and returns a slice of up to n FileInfo values, as would be returned by Lstat, **in directory order**.

Running tests on this package on a CentOS 6.6 machine with an ext3 filesystem produces this failure.

```
--- FAIL: TestConfig_LoadConfigsFileOrder (0.00s)
	config_test.go:331: Loaded configs don't match
		want: [test-resources/etcnomad/common.hcl test-resources/etcnomad/server.json test-resources/myconf]
		 got: [test-resources/etcnomad/server.json test-resources/etcnomad/common.hcl test-resources/myconf]
```

--- a/command/agent/config.go
+++ b/command/agent/config.go
@@ -8,6 +8,7 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"sort"
 	"strings"
 	"time"
 
@@ -704,6 +705,8 @@ func LoadConfigDir(dir string) (*Config, error) {
 	if len(files) == 0 {
 		return &Config{}, nil
 	}
+
+	sort.Strings(files)
 
 	var result *Config
 	for _, f := range files {

